Mathlib/Algebra/Order/Monovary.lean

```diff
@@ -0,0 +1,428 @@
+
+## See also
+
+`Algebra.Order.Rearrangement` for the n-ary rearrangement inequality
+-/
+
+variable {ι α β : Type*}
+
+/-! ### Algebraic operations on monovarying functions -/
+
+section OrderedCommGroup
+variable [OrderedCommGroup α] [OrderedCommGroup β] {s : Set ι} {f f₁ f₂ : ι → α} {g g₁ g₂ : ι → β}
+
+@[to_additive (attr := simp)]
+lemma monovaryOn_inv_left : MonovaryOn f⁻¹ g s ↔ AntivaryOn f g s := by
+  simp [MonovaryOn, AntivaryOn]
+
+@[to_additive (attr := simp)]
+lemma antivaryOn_inv_left : AntivaryOn f⁻¹ g s ↔ MonovaryOn f g s := by
+  simp [MonovaryOn, AntivaryOn]
+
+@[to_additive (attr := simp)]
+lemma monovaryOn_inv_right : MonovaryOn f g⁻¹ s ↔ AntivaryOn f g s := by
+  simpa [MonovaryOn, AntivaryOn] using forall₂_swap
+
+@[to_additive (attr := simp)]
+lemma antivaryOn_inv_right : AntivaryOn f g⁻¹ s ↔ MonovaryOn f g s := by
+  simpa [MonovaryOn, AntivaryOn] using forall₂_swap
+
+@[to_additive] lemma monovaryOn_inv : MonovaryOn f⁻¹ g⁻¹ s ↔ MonovaryOn f g s := by simp
+@[to_additive] lemma antivaryOn_inv : AntivaryOn f⁻¹ g⁻¹ s ↔ AntivaryOn f g s := by simp
+
+@[to_additive (attr := simp)] lemma monovary_inv_left : Monovary f⁻¹ g ↔ Antivary f g := by
+  simp [Monovary, Antivary]
+
+@[to_additive (attr := simp)] lemma antivary_inv_left : Antivary f⁻¹ g ↔ Monovary f g := by
+  simp [Monovary, Antivary]
+
+@[to_additive (attr := simp)] lemma monovary_inv_right : Monovary f g⁻¹ ↔ Antivary f g := by
+  simpa [Monovary, Antivary] using forall_swap
+
+@[to_additive (attr := simp)] lemma antivary_inv_right : Antivary f g⁻¹ ↔ Monovary f g := by
+  simpa [Monovary, Antivary] using forall_swap
+
+@[to_additive] lemma monovary_inv : Monovary f⁻¹ g⁻¹ ↔ Monovary f g := by simp
+@[to_additive] lemma antivary_inv : Antivary f⁻¹ g⁻¹ ↔ Antivary f g := by simp
+
+@[to_additive] alias ⟨MonovaryOn.of_inv_left, AntivaryOn.inv_left⟩ := monovaryOn_inv_left
+@[to_additive] alias ⟨AntivaryOn.of_inv_left, MonovaryOn.inv_left⟩ := antivaryOn_inv_left
+@[to_additive] alias ⟨MonovaryOn.of_inv_right, AntivaryOn.inv_right⟩ := monovaryOn_inv_right
+@[to_additive] alias ⟨AntivaryOn.of_inv_right, MonovaryOn.inv_right⟩ := antivaryOn_inv_right
+@[to_additive] alias ⟨MonovaryOn.of_inv, MonovaryOn.inv⟩ := monovaryOn_inv
+@[to_additive] alias ⟨AntivaryOn.of_inv, AntivaryOn.inv⟩ := antivaryOn_inv
+@[to_additive] alias ⟨Monovary.of_inv_left, Antivary.inv_left⟩ := monovary_inv_left
+@[to_additive] alias ⟨Antivary.of_inv_left, Monovary.inv_left⟩ := antivary_inv_left
+@[to_additive] alias ⟨Monovary.of_inv_right, Antivary.inv_right⟩ := monovary_inv_right
+@[to_additive] alias ⟨Antivary.of_inv_right, Monovary.inv_right⟩ := antivary_inv_right
+@[to_additive] alias ⟨Monovary.of_inv, Monovary.inv⟩ := monovary_inv
+@[to_additive] alias ⟨Antivary.of_inv, Antivary.inv⟩ := antivary_inv
+
+@[to_additive] lemma MonovaryOn.mul_left (h₁ : MonovaryOn f₁ g s) (h₂ : MonovaryOn f₂ g s) :
+    MonovaryOn (f₁ * f₂) g s := fun _i hi _j hj hij ↦ mul_le_mul' (h₁ hi hj hij) $ h₂ hi hj hij
```

Comment:
```suggestion
    MonovaryOn (f₁ * f₂) g s := fun _i hi _j hj hij ↦ mul_le_mul' (h₁ hi hj hij) <| h₂ hi hj hij
```

Reply:
and below. I think `mul_le_mul' (h₁ hi hj hij) (h₂ hi hj hij)` looks nicer here because of the symmetry, but that's up to you.